### ISSUE TITLE
Contributing to the project

### ISSUE THREAD

**Issue by dyba** [JodaOrg/joda-time#234]
Hi!

I'm interested in contributing to the project. How do I get the test suite to run? I'm trying to set up the configuration on IntelliJ, but I'm missing a step in the configuration that points to the resources folder:

```
java.io.IOException: Resource not found: "org/joda/time/tz/data/ZoneInfoMap" ClassLoader: sun.misc.Launcher$AppClassLoader@77fe0d66
```
![screen shot 2015-01-30 at 14 52 26](https://cloud.githubusercontent.com/assets/296219/5984885/a819bb0a-a88f-11e4-8d64-2a30636e78b4.png)

Thanks!

**Comment by dyba** [comment]
I did find [this link](http://sourceforge.net/p/joda-time/discussion/337835/thread/2798a578/) to a discussion about the same issue; however, it is strange to me that when I run all the tests instead of just the one test class I introduced, the error still appears.

**Comment by dyba** [comment]
I narrowed the problem to [this line](https://github.com/JodaOrg/joda-time/blob/master/src/main/java/org/joda/time/tz/ZoneInfoProvider.java#L202) in the source code. At this point in time, with my configurations, ```path``` is equal to ```"org/joda/time/tz/data/ZoneInfoMap"```, but actually it should be pointing to ```"src/test/resources/tzdata/ZoneInfoMap"```.

**Comment by jodastephen** [comment]
In Eclipse, I compile the time-zone data using maven and then move the data to a folder that is on the classpath under the path specified. Thus I have three classpath folders, one for the source `src/main/java`, one for tests `src/test/java` and one for the time-zone data `src/tz-data` (where the files are in the subfolder/package `org/joda/time/tz/data`).

**Comment by dyba** [comment]
Thanks @jodastephen ! I hope you don't mind me asking further questions about this. I'm new to the Java environment. What do you mean you compile the time-zone data using maven? This is probably where I create a Maven Run configuration in my IntelliJ IDE perhaps and then point that configuration to the correct folder? And are you talking about the time zone data in ```src/test/resources/tzdata```? Or the time zone data in ```src/main/java/org/joda/time/tz/src```?

So it seems that after I get passed that compilation part, I will have to create a new folder ```src/tz-data``` and placed the compiled file or files in this new folder?

Thanks for taking the time to respond!

**Comment by dyba** [comment]
I think I got further along now. I was able to compile the time zone data using these settings:

![screen shot 2015-02-02 at 12 04 42](https://cloud.githubusercontent.com/assets/296219/6008098/c54e56ea-aad3-11e4-978f-137f25c828f7.png)

And it produced a data folder in ```target/classes/org/joda/time/tz/```. I don't know how to modify my classpaths, but I'll look into it. I suppose from here, according to your instructions, I move that ```data``` folder I mentioned and drop the files in that folder in a new folder ```src/tz-data```.

I am getting a new error, but this is probably because I haven't moved the time zone data to the classpath:

```
java.lang.RuntimeException: Failed to invoke suite(): class org.joda.time.chrono.TestAll java.lang.ExceptionInInitializerError
	at com.intellij.junit3.TestRunnerUtil$FailedTestCase.runTest(TestRunnerUtil.java:243)
	at com.intellij.junit3.JUnit3IdeaTestRunner.doRun(JUnit3IdeaTestRunner.java:141)
	at com.intellij.junit3.JUnit3IdeaTestRunner.startRunnerWithArgs(JUnit3IdeaTestRunner.java:52)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:211)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at com.intellij.junit3.TestRunnerUtil.createClassOrMethodSuite(TestRunnerUtil.java:122)
	at com.intellij.junit3.TestRunnerUtil.getTestSuite(TestRunnerUtil.java:81)
	at com.intellij.junit3.JUnit3IdeaTestRunner.startRunnerWithArgs(JUnit3IdeaTestRunner.java:50)
	... 7 more
Caused by: java.lang.ExceptionInInitializerError
	at org.joda.time.chrono.TestAll.suite(TestAll.java:51)
	... 14 more
Caused by: java.lang.IllegalArgumentException: The datetime zone id 'Europe/Paris' is not recognised
	at org.joda.time.DateTimeZone.forID(DateTimeZone.java:229)
	at org.joda.time.chrono.TestDays360Chronology.<clinit>(TestDays360Chronology.java:29)
	... 15 more
```

**Comment by dyba** [comment]
::FacePalm:: I forgot that I had changed a line of code in ```DateTimeZone.java``` that prevented the application from finding the newly compiled time zone data. All is working well. Thanks @jodastephen !

By the way, I would like to use the information in this thread and provide a tutorial for newer developers who are interested in contributing to the project. Perhaps the project can have a CONTRIB.md file? What do you think?

**Comment by jodastephen** [comment]
Adding a `CONTRIBUTING.md` file would be OK, although relatively few people actually contribute!

**Comment by dyba** [comment]
Well, I hope at least those who are interesting in contributing won't get hung up with running the test cases. I'll draft up a markdown doc with that filename and send a PR.

Incidentally, I did try my hand at creating a 360-day chronology but realized it was more tedious than I had expected. When I found out I had to keep track of milliseconds to offset dates, I decided I would try to build a 360-days function in my target language instead, Clojure. It proved much easier. :wink:
